pkg/git/gitlab/gitlab_helper.go

```diff
@@ -42,6 +67,24 @@ func getProjectPathFromRepoUrl(repoUrl string) (string, error) {
 	), nil
 }
 
+func GetBaseUrl(repoUrl string) (string, error) {
+	url, err := url.Parse(repoUrl)
+	if err != nil {
+		return "", FailedToParseUrlError{url: repoUrl, err: err.Error()}
```

Comment:
Custom errors looks redundant here: they are never used / distinguished.

Reply:
they are used in the tests :)